PWGGA/GammaConvBase/AliConversionMesonCuts.cxx

```diff
@@ -5047,3 +5073,72 @@ Bool_t AliConversionMesonCuts::IsParticleInJet(std::vector<Double_t> vectorJetEt
 
   return particleInJet;
 }
+
+
+    
+
+///_____________________________________________________________________________
+//   Function to find highest pT track in event
+template <class T, class U>
+Bool_t AliConversionMesonCuts::MesonLeadTrackSelection(T Event, U meson)
+{
+  if (fInLeadTrackDir == 0) return kTRUE;
+
+  if(!meson) {
+    AliError("Meson not available\n");
+    return kFALSE;
+  }  
+  if(!Event) {
+    AliError("Event not available\n");
+    return kFALSE;
+  }
+
+  Double_t PtLead = 0;
+  Int_t NTotTrack = 0;
+  Double_t Selection = TMath::Pi()/3.; // 120 grad
+
+  AliVTrack* track = nullptr;
+  AliVTrack* trackLead = nullptr;
+
+  for(int i = 0; i < Event->GetNumberOfTracks(); i++)
+  {
+    track = dynamic_cast<AliVTrack*>(Event->GetTrack(i));
+    if(!track) {
+      AliError("Track not available\n");
+      continue;
+    }
+    // TODO: add quality cuts????
+    // apply min pT cut for tracks
+    if(track->Pt() > fLeadTrackMinPt){
+      // Find track with highest Pt
+      if(track->Pt() > PtLead)
+      {
+        PtLead  = track->Pt();
+        trackLead = dynamic_cast<AliVTrack*>(Event->GetTrack(i));
+      }
+      NTotTrack  += 1;
+    }
+  }
+
+  if (NTotTrack > 0) {
+    // get meson phi angle
+    TVector3 vmeson(meson->Px(),meson->Py(),meson->Pz());
+    TVector3 vlead(trackLead->Px(), trackLead->Py(), trackLead->Pz());
+    Double_t AngleToLeadTrack = vmeson.Angle(vlead); // this is in rad
+    //  in lead track direction
+    if(fInLeadTrackDir == 1){  
+      if ( AngleToLeadTrack < Selection/2 ) return kTRUE;
+    //  transwerse to lead track direction
+    } else if(fInLeadTrackDir == 2){
+      if ( AngleToLeadTrack > Selection/2 && AngleToLeadTrack < Selection ) return kTRUE;
+    //  opposite direction to lead track (away)
+    } else if(fInLeadTrackDir == 3){
+      if ( AngleToLeadTrack > Selection && AngleToLeadTrack < TMath::Pi() ) return kTRUE;
+    //  in direction and opposite direction
+    } else if(fInLeadTrackDir == 4){
+      if (( AngleToLeadTrack > Selection/2 && AngleToLeadTrack < Selection ) || ( AngleToLeadTrack > Selection && AngleToLeadTrack < TMath::Pi() )) return kTRUE;
+    }
```

Comment:
The angle you get from TVector3 Angle should be the geometric angle between the two 3D vectors, not the phi vector as you state in the comment. I assume you meant the phi vector because afaik in Kristina's thesis she just cut on phi but ignored eta. So you would need to compare T3Vector::Phi() for both particles I think.

Reply:
thats correct, I am changing it to be only phi so we can better relate to that analysis